composer/helpers/v1/src/UpdateChecker.php

```diff
@@ -82,6 +82,11 @@ public static function getLatestResolvableVersion(array $args): ?string
 
         // We found the package in the list of updated packages. Return its version.
         if ($updatedPackage instanceof PackageInterface) {
+            // TODO surprisingly the returned result of getPrettyVersion depends on the PHP version:
+            // - PHP 7 returns: "2.4.1"
+            // - PHP 8 returns: "2.4.1@stable"
+            file_put_contents('php://stdout', $updatedPackage->getPrettyVersion());
```

Comment:
@Seldaek any idea why the returned value of `getPrettyVersion()` version depends on the underlying PHP version? I found this very surprising given that the composer code didn't change, only the php version.

This is a unit test running under `composer` `1`, accessing a PEAR registry.

I do realize that both PEAR and composer v1 are deprecated, but we're not _quite_ ready to drop support for them, so I'd like to fix our code to buy us compatibility for a little longer.

And really what I care about is:
Is this `@stable` suffix always going to be the same returned value such that it's safe for me to simply strip it off? Or could it sometimes return `@beta`, `-dev`, etc? 

IE, is `@stable` something PHP itself is adding in some manner in which case I'd always expect it to be there and I can safely strip it (I don't know if PHP has a concept of stable/unstable objects aka immutable?), or is it something `composer` is adding, so it may vary depending on the package/registry?